src/widget/whotcuebutton.cpp

```diff
@@ -0,0 +1,150 @@
+
+    auto pDisplayConnection = new ControlParameterWidgetConnection(
+            this,
+            createConfigKey(QStringLiteral("enabled")),
+            nullptr,
+            ControlParameterWidgetConnection::DIR_TO_WIDGET,
+            ControlParameterWidgetConnection::EMIT_NEVER);
+    addConnection(pDisplayConnection);
+    setDisplayConnection(pDisplayConnection);
+
+    QDomNode con = context.selectNode(node, QStringLiteral("Connection"));
+    if (!con.isNull()) {
+        SKIN_WARNING(node, context) << "Additional Connections are not allowed";
+    }
+}
+
+void WHotcueButton::mousePressEvent(QMouseEvent* e) {
+    const bool rightClick = e->button() == Qt::RightButton;
+    if (rightClick) {
+        if (readDisplayValue() == 1) {
+            // hot cue is set
+            TrackPointer pTrack = PlayerInfo::instance().getTrackInfo(m_group);
+            if (!pTrack) {
+                return;
+            }
+
+            CuePointer pHotCue;
+            QList<CuePointer> cueList = pTrack->getCuePoints();
+            for (const auto& pCue : cueList) {
+                if (pCue->getHotCue() == m_hotcue) {
+                    pHotCue = pCue;
+                    break;
+                }
+            }
+            if (!pHotCue) {
+                return;
+            }
+            m_pCueMenuPopup->setTrackAndCue(pTrack, pHotCue);
+            m_pCueMenuPopup->popup(e->globalPos());
+        }
+        return;
+    }
+
+    // Pass all other press events to the base class.
+    WPushButton::mousePressEvent(e);
+}
+
+ConfigKey WHotcueButton::createConfigKey(const QString& name) {
+    ConfigKey key;
+    key.group = m_group;
+    // Add one to hotcue so that we don't have a hotcue_0
+    key.item = QStringLiteral("hotcue_") + QString::number(m_hotcue + 1) + QChar('_') + name;
+    return key;
+}
+
+void WHotcueButton::slotColorChanged(double color) {
+    if (color < 0 && color > 0xFFFFFF) {
+        return;
+    }
+    m_cueColor = QColor::fromRgb(color);
+    updateStyleSheet();
+}
+
+void WHotcueButton::setLightTextColorChanged(QColor color) {
+    if (m_lightTextColor != color) {
+        m_lightTextColor = color;
+        updateStyleSheet();
+    }
+}
+
+void WHotcueButton::setDarkTextColorColorChanged(QColor color) {
+    if (m_darkTextColor != color) {
+        m_darkTextColor = color;
+        updateStyleSheet();
+    }
+}
+
+void WHotcueButton::updateStyleSheet() {
+    if (!m_cueColor.isValid()) {
+        return;
+    }
+    QColor textColor = Color::isDimmColor(m_cueColor) ? m_lightTextColor : m_darkTextColor;
+
+    QString style =
+            QStringLiteral("WWidget[displayValue=\"1\"] { background-color: ") +
+            m_cueColor.name() +
+            (textColor.isValid() ? (QStringLiteral("; color: ") +
+                                           textColor.name())
+                                 : QString()) +
+            QStringLiteral("; }");
+
+    if (m_hoverCueColor) {
```

Comment:
I will give it a try.

Reply:
palette() is in this case the default QPalette() set by QApplication::setPalette(). 
We may change it but this restyles the whole Application, so this is no option yet.

Reply:
> palette() is in this case the default QPalette() set by QApplication::setPalette().
> We may change it but this restyles the whole Application, so this is no option yet.

Ok, that unfortunate. @ronso0 Do you have any ideas how to proceed? Do you think these two properies are sufficient?

Reply:
Actually I'm not happy with just _light_ and _dark_. For the 'white border' example below it would be more convenient to have _dark_, _medium_ & _bright_ so that I could use the white border for _dark_ and _medium_ and only change it to grey if we have a _bright_ hotcue.

Nevertheless, I think this is okay fo now. Also I think brightness() is used elsewhere in Mixxx, isn't it?